### DISCUSSION TITLE
ChromeNotInstalledError within Windows VM

### DISCUSSION THREAD

**Discussion by ruskspace** [GoogleChrome/lighthouse#12618] [Q&A] [answered]
#### What is the current behavior?

When I run my node app using this example in a Windows Datacenter 2019 VM that has Node and Chrome it produces a ChromeNotInstalledError: https://github.com/GoogleChrome/lighthouse/blob/master/docs/readme.md

If I run this before starting node it executes, but it doesn't maintain once I stop node:
`$env:CHROME_PATH = (Get-Process -Name chrome)[0].Path`

The same app running locally on my laptop with Chrome executes as expected.

#### What is the expected behavior?

For the Windows VM running my app to produce the lighthouse report akin to how it runs on my laptop that has the same Node, Chrome, and Lighthouse versions.

#### Environment Information
* Affected Channels: Using Lighthouse programmatically with Node Module
* Lighthouse version: 8.0.0
* Chrome version: Version 91.0.4472.77 (Official Build) (64-bit)
* Node.js version: v14.17.0
* Operating System: Windows Datacenter 2019 VM

**Comment by patrickhulce** [comment, marked as answer]
@ruskspace if you're running programmatically then you can use the [`chromePath` option of chrome-launcher](https://github.com/GoogleChrome/chrome-launcher/#launch-options), this purpose is exactly what it's there for :) 

```js
await chromeLauncher.launch({chromePath: getMyCustomChromePath()})
```

**Reply by ruskspace** [reply]
Thanks. Does that work with the ChromeFlags set to headless? Would I comma deliminate both of the options?

`await chromeLauncher.launch({chromeFlags: ['--headless']});`

And any ideas why it works on my laptop but not in the VM without needing to declare the ChromePath

**Reply by patrickhulce** [reply]
Yep you can use both

`await chromeLauncher.launch({chromeFlags: ['--headless'], chromePath: getMyCustomChromePath()});`

> And any ideas why it works on my laptop but not in the VM without needing to declare the ChromePath

Chrome launcher automatically finds a few common user locations for running on local machines, but the range of possible locations on VMs is usually much broader (so not covered) and explicit declaration is the expected usage.

**Reply by ruskspace** [reply]
I tried the following, but it is producing the same issue:

`const chrome = await chromeLauncher.launch({chromeFlags: ['--headless'], chromePath: "C:\Program Files\Google\Chrome\Application\chrome.exe"});`

If this doesn't work, since setting this first before starting node works, is there a way to set this in the Node app to always run?

`$env:CHROME_PATH = (Get-Process -Name chrome)[0].Path`

**Reply by ruskspace** [reply]
Got it working. I set an Windows Environmental Variable of CHROME_PATH with a value of C:\Program Files\Google\Chrome\Application\chrome.exe

Once I restarted the VM it picked it up.
